idanalysis/eqparams.py

```diff
@@ -0,0 +1,1294 @@
+        i2y = _np.trapz(dx=ds, y=(bx / self.brho) ** 2)
+        self._i2x = i2x
+        self._i2y = i2y
+        return i2x, i2y
+
+    def calc_i3(self):
+        """Calculate third radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I3x
+            float: delta I3y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        s = 1e-3 * self.field_profile[:, 0]
+        ds = _np.diff(s)[0]
+        i3x = _np.trapz(dx=ds, y=_np.abs((by / self.brho) ** 3))
+        i3y = _np.trapz(dx=ds, y=_np.abs((bx / self.brho) ** 3))
+        self._i3x = i3x
+        self._i3y = i3y
+        return i3x, i3y
+
+    def calc_i4(self):
+        """Calculate fourth radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I4x
+            float: delta I4y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        s = 1e-3 * self.field_profile[:, 0]
+        ds = _np.diff(s)[0]
+        dby = _np.diff(by)
+        dbx = _np.diff(bx)
+        dby_ds = _np.append(dby, dby[-1])/ds
+        dbx_ds = _np.append(dbx, dbx[-1])/ds
+        kx = -cumtrapz(dx=ds, y=by*dby_ds, initial=0) / (self.brho)**2
+        ky = -cumtrapz(dx=ds, y=-bx*dbx_ds, initial=0) / (self.brho)**2
+        etax, etay, *_ = self.calc_dispersion()
+        i4x = _np.trapz(
+            dx=ds,
+            y=etax * (by / self.brho) ** 3 - 2 * ky * etax * (by / self.brho),
+        )
+        i4y = _np.trapz(
+            dx=ds,
+            y=etay * (bx / self.brho) ** 3 - 2 * kx * etay * (bx / self.brho),
+        )
+        self._i4x = i4x
+        self._i4y = i4y
+        return i4x, i4y
+
+    def calc_i5(self):
+        """Calculate fifth radiation integral contribution from one ID.
+
+        Returns:
+            float: delta I5x
+            float: delta I5y
+        """
+        by = self.field_profile[:, 1]
+        bx = self.field_profile[:, 2]
+        s = 1e-3 * self.field_profile[:, 0]
+        ds = _np.diff(s)[0]
+        hx, hy = self.get_curly_h()
+        i5x = _np.trapz(dx=ds, y=hx * _np.abs((by / self.brho) ** 3))
+        i5y = _np.trapz(dx=ds, y=hy * _np.abs((bx / self.brho) ** 3))
+        self._i5x = i5x
+        self._i5y = i5y
+        return i5x, i5y
+
+
+class EqParamAnalysis:
+    """Class to calculate beam equilibrium parameters.
+
+    All equations used here are from the book: Clarke, James A.,
+    'The Science and Technology of Undulators and Wigglers'.
+    Chapter 10: 'The Effect of Insertion Devices on the Electron Beam'.
+    """
+
+    def __init__(self, beam_energy=3.0):
+        """Class constructor.
+
+        Args:
+            beam_energy (Float, optional): Beam energy [GeV]. Defaults to 3.0.
+        """
+        self.ids = list()
+        self._model = _model
+        self._eq_params_nominal = None
+        self._beam = Beam(beam_energy)
+        self.brho = self._beam.brho
```

Comment:
why do you need to replicate these symbols? couldn't you use `self._beam.brho` when you need them?

Reply:
Yes, we could use self._beam.brho, but using just self.brho saves space, and makes the code a little bit more readable.

Reply:
For now this is not a problem, because you didn't define a setter for the beam energy. But this is a potential problem in the future in case we do that. Duplicating symbols is a potential bug, I think it is better not to do that.
If the problem is the length of the line where you will use it, you can simply define a variable for it in the previous line, like this: `brho = self._beam.brho`. It is even shorter.